contracts/misc/RewardSplitter.sol

```diff
@@ -134,10 +159,75 @@ contract RewardSplitter is IRewardSplitter, Initializable, OwnableUpgradeable, M
     uint256 rewards,
     address receiver
   ) external override returns (uint256 positionTicket) {
+    if (rewards == type(uint256).max) {
+      rewards = rewardsOf(msg.sender);
+    }
     _withdrawRewards(msg.sender, rewards);
     return IVaultEnterExit(vault).enterExitQueue(rewards, receiver);
   }
 
+  /// @inheritdoc IRewardSplitter
+  function enterExitQueueOnBehalf(uint256 rewards, address onBehalf) external {
+    if (!isClaimOnBehalfEnabled) revert Errors.AccessDenied();
+
+    if (rewards == type(uint256).max) {
+      rewards = rewardsOf(onBehalf);
+    }
+    _withdrawRewards(onBehalf, rewards);
```

Comment:
After syncRewards the rewardsOf can return different value. You must make sure to get the max possible amount after the state is up to date.